### DISCUSSION TITLE
Wrapping an image around a 3D object

### DISCUSSION THREAD

**Discussion by ivanamar996** [ImageMagick/ImageMagick#6322] [Consulting]
Hi,

I am trying to wrap an image around the object by using command:

magick MO8933-03_POS1_vectormap.png -resize "2695x2710^!" MO8933-1_tex_map.png -virtual-pixel Edge -compose Distort -define compose:args=100%%x100%% -composite MO8933-03_POS1.jpg -compose Overlay -composite pntp_usr.png

I have created vector mask that represent the are on which logo should be placed.

The mask : 
![MO8933-03_POS1_vectormap](https://github.com/ImageMagick/ImageMagick/assets/39298540/5d9e1f33-1964-4178-921e-1b7028445bc5)

Artwotk: 

![MO8933-1_tex_map](https://github.com/ImageMagick/ImageMagick/assets/39298540/21b39346-bbad-4af0-8a2f-8459fd25995d)

and the product image is:

![MO8933-03_POS1](https://github.com/ImageMagick/ImageMagick/assets/39298540/27a77e53-0d4a-4703-880b-8960389059f9)

But for some reason the end result is not good: 

![pntp_usr](https://github.com/ImageMagick/ImageMagick/assets/39298540/6d30d53b-0476-488a-8363-60e2a841807f)

Do you maybe know what I am doing wrong? The issue is on step that am trying to create image based on vector map and artwork, but for some reason it ignores map and all colors from the logo disappears and are converted to black. 

I am using image magick version:

Version: ImageMagick 7.1.0-55 Q16-HDRI x64 6edfae3:20221217 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193431937)

Kind regads,
Ivana

**Comment by snibgo** [comment]
What is your platform? I assume Windows BAT.

I can't see anything wrong in your command. It works fine for me, with IM v7.1.0-62.
```
magick ^
  mArtwork.png ^
  -resize "2695x2710^!" ^
  mMap.png ^
  -virtual-pixel Edge ^
  -compose Distort -define compose:args=100%%x100%% -composite ^
  mProduct.jpg ^
  -compose Overlay -composite ^
  x.png
```

![x](https://github.com/ImageMagick/ImageMagick/assets/33812368/348c16fb-7bd6-4196-ac1c-505ec9aae411)

If your platform isn't Windows BAT, that might explain the problem.

The quality can be improved by supersampling:
```
magick ^
  mArtwork.png ^
  -resize "2695x2710^!" ^
  mMap.png ^
-resize 200%% ^
  -virtual-pixel Edge ^
  -compose Distort -define compose:args=100%%x100%% -composite ^
-resize 50%% ^
  mProduct.jpg ^
  -compose Overlay -composite ^
  x2.png
```

EDIT: Another general point: your map file is 8-bit per channel PNG. This means the general error in displacement will be 1 part in 512, which is very visible. Displacement maps need to be high precision. I suggest at least 16 bits per channel, or even 32 bits either floating-point or integer. (PNG files are limited to 16 bits. TIFF files can be 32 bits.)
